### PR TITLE
refactor(RegExpWrapper): remove the facade

### PR DESCRIPTION
**Pull request by jasonchoimtt**
based on @vicb 's work. can you review?

**Please check if the PR fulfills these requirements**
- The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
Refactoring (no functional changes, no api changes)
```

**Does this PR introduce a breaking change?** (check one with "x")

```
No
```

### CODE REVIEW

File: modules/@angular/compiler/src/style_url_resolver.ts [hunk 1]
```diff
@@ -45,4 +45,4 @@ export function extractStyleUrls(
 var _cssImportRe = /@import\s+(?:url\()?\s*(?:(?:['"]([^'"]*))|([^;\)\s]*))[^;]*;?/g;
 // TODO: can't use /^[^:/?#.]+:/g due to clang-format bug:
 //       https://github.com/angular/angular/issues/4596
-var _urlWithSchemaRe = /^([a-zA-Z\-\+\.]+):/g;
+var _urlWithSchemaRe = /^([a-zA-Z\-\+\.]+):/;
```
Comment (vicb) [inline modules/@angular/compiler/src/style_url_resolver.ts]: could you please switch the regexp given in the comment above and remove the comment. Thanks

**Comment by vicb** [comment]
Really good job, Thanks !

Could you please address the comment and fix the formatting.

Feel free to add the "merge" label once done.

**Comment by jasonchoimtt** [comment]
Turns out the regexp in the comment is incorrect; I changed to `/^([^:/?#]+):/`. (without dot and with parentheses).

Please help me add the label as I can't do that. :)

File: modules/@angular/compiler/src/selector.ts [hunk 2]
```diff
@@ -8,17 +8,18 @@
...
-    '(\\))|' +                                // ")"
-    '(\\s*,\\s*)');                           // ","
+const _SELECTOR_REGEXP = new RegExp(
+    '(\\:not\\()|' +                              //":not("
+        '([-\\w]+)|' +                            // "tag"
+        '(?:\\.([-\\w]+))|' +                     // ".class"
+        '(?:\\[([-\\w*]+)(?:=([^\\]]*))?\\])|' +  // "[name]", "[name=value]" or "[name*=value]"
+        '(\\))|' +                                // ")"
+        '(\\s*,\\s*)',
+    'g');  // ","
```
Comment (vicb) [inline modules/@angular/compiler/src/selector.ts]: the comment should be on the previous line

File: modules/@angular/router-deprecated/src/url_parser.ts [hunk 3]
```diff
@@ -89,15 +89,15 @@ export function pathSegmentsToUrl(pathSegments: string[]): Url {
   return url;
 }
 
-var SEGMENT_RE = RegExpWrapper.create('^[^\\/\\(\\)\\?;=&#]+');
+const SEGMENT_RE = /^[^\/\(\)\\?;=&#]+/;
```
Comment (vicb) [inline modules/@angular/router-deprecated/src/url_parser.ts]: should be `\?` here

**Comment by vicb** [comment]
Added a couple of comments, I'll label when travis is green

**Comment by vicb** [comment]
Waouh that was fast, LGTM please squash

**Pull request merged**